src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2015,13 +2015,30 @@ public override BoundNode VisitReturnStatement(BoundReturnStatement node)
             }
         }
 
-        private BetterResult BetterConversionTarget(TypeSymbol type1, TypeSymbol type2, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        private BetterResult BetterConversionTarget(
+            TypeSymbol type1,
+            TypeSymbol type2,
+            ref HashSet<DiagnosticInfo> useSiteDiagnostics,
+            int betterConversionTargetRecursionLimit = 100)
```

Comment:
This current pattern makes it easy to forget to pass the recursion thus rendering it useless.  If the separation between outside and inside overload resolution is teh goal then divide the methods:

1. BetterConversionTarget: called outside overload resolution, no need for a recursion argument.
2. BetterConversionTargetCore: called inside, pass the recursion limit.